tests/test_feedexport.py

```diff
@@ -186,6 +186,147 @@ def test_store(self):
         content = get_s3_content_and_delete(u.hostname, u.path[1:])
         self.assertEqual(content, expected_content)
 
+    def test_init_without_acl(self):
+        storage = S3FeedStorage(
+            's3://mybucket/export.csv',
+            'access_key',
+            'secret_key'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, None)
+
+    def test_init_with_acl(self):
+        storage = S3FeedStorage(
+            's3://mybucket/export.csv',
+            'access_key',
+            'secret_key',
+            'custom-acl'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, 'custom-acl')
+
+    def test_from_crawler_without_acl(self):
+        settings = {
+            'AWS_ACCESS_KEY_ID': 'access_key',
+            'AWS_SECRET_ACCESS_KEY': 'secret_key',
+        }
+        crawler = get_crawler(settings_dict=settings)
+        storage = S3FeedStorage.from_crawler(
+            crawler,
+            's3://mybucket/export.csv'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, None)
+
+    def test_from_crawler_with_acl(self):
+        settings = {
+            'AWS_ACCESS_KEY_ID': 'access_key',
+            'AWS_SECRET_ACCESS_KEY': 'secret_key',
+            'FEED_STORAGE_S3_ACL': 'custom-acl',
+        }
+        crawler = get_crawler(settings_dict=settings)
+        storage = S3FeedStorage.from_crawler(
+            crawler,
+            's3://mybucket/export.csv'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, 'custom-acl')
+
+    @defer.inlineCallbacks
+    def test_store_botocore_without_acl(self):
+        if os.getenv('TOXENV') == 'jessie':
```

Comment:
What do you think about skipping a test if botocore is not installed?

Reply:
If someone accidentally removes botocore from other test environment requirements we could end up skipping this test and therefore hiding some errors.

Leaving it as it is now is also a good point for when someone is removing the jessie support. A quick grep on the source code would highlight this conditional statement and the exception raise line.

Reply:
I think that it is better to skip the test if botocore is not present because of two reasons:

1. botocore is an optional Scrapy requirement; we should be running tests without it, to make sure Scrapy works without botocore. So it is fine not to have botocore in an environment.
2. People are running Scrapy tests without tox, e.g. on their environments; tests currently are not aware of tox and don't require it - checking for TOXENV variable changes that.

Which methods are missing in botocore? Should we specify a minimal botocore version somewhere (`is_botocore` function? docs?)

Reply:
Ok, I'll just check the `is_botocore` method then. 

About the other question, I did some research and this is what I found out:

- the botocore version specified in Debian Jessie page is 0.62 (https://github.com/scrapy/scrapy/pull/3607#discussion_r256587948)
- the `Session.create_client` method appeared in version 0.64 (https://github.com/boto/botocore/commit/dd5aad504fd6bbe13e327460f10fbac96cc632b9)
- the `aws_access_key_id` and `aws_secret_access_key` args were introduced in 0.66 (https://github.com/boto/botocore/commit/59a1c336881a618eb321e317ea70beae81017836)
- there was a bug with `_send_request` receive an extra argument solved in 0.69 (https://github.com/boto/botocore/commit/a84a06df9e1df64f1ac96b0b54a1319f6c89881b)

Those versions are pretty old so I believe we shouldn't have problems.